sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -106,21 +106,31 @@ private[hive] class HiveClientImpl(
 
     // Set up kerberos credentials for UserGroupInformation.loginUser within
     // current class loader
-    // Instead of using the spark conf of the current spark context, a new
-    // instance of SparkConf is needed for the original value of spark.yarn.keytab
-    // and spark.yarn.principal set in SparkSubmit, as yarn.Client resets the
-    // keytab configuration for the link name in distributed cache
     if (sparkConf.contains("spark.yarn.principal") && sparkConf.contains("spark.yarn.keytab")) {
       val principalName = sparkConf.get("spark.yarn.principal")
-      val keytabFileName = sparkConf.get("spark.yarn.keytab")
-      if (!new File(keytabFileName).exists()) {
-        throw new SparkException(s"Keytab file: ${keytabFileName}" +
-          " specified in spark.yarn.keytab does not exist")
-      } else {
-        logInfo("Attempting to login to Kerberos" +
-          s" using principal: ${principalName} and keytab: ${keytabFileName}")
-        UserGroupInformation.loginUserFromKeytab(principalName, keytabFileName)
+      val keytabFileName = {
+        val keytab = sparkConf.get("spark.yarn.keytab")
+        if (new File(keytab).exists()) {
+          keytab
+        } else {
+          // In case user still uses old SparkConf (reset by yarn#client to point to location in
```

Comment:
I'm not sure I understand the comment here. This doesn't have anything to do with the user creating an old SparkConf?  Isn't this just in yarn client mode it has to use the original SparkConf where yarn.Client didn't update the location of the keytab file?  If that is the case perhaps just update the comment to say something to that affect.